Add support for interactive option

BrunnerLivio opened the pull request:
Adds support for `--interactive` option

Examples:

```bash
ts-node --eval "const hello = 'world';" --interactive
> console.log(hello)
world
undefined
```

```bash
ts-node --eval "import * as react from 'react';" --interactive
> react.version
'16.2.0'
```

BrunnerLivio commented:
This PR is quite possibly pretty hacky. Tried my best to reach the described examples. I have already found some issues:

```bash
➜ ts-node src/bin.ts --eval "const hello = 'world'" --interactive 
> console.log(hello)
[eval].ts:1:22 - error TS2451: Cannot redeclare block-scoped variable 'console'.

1 const hello = 'world'console.log(hello)
                       ~~~~~~~

  node_modules/@types/node/globals.d.ts:164:13
    164 declare var console: Console;
                    ~~~~~~~
    'console' was also declared here.
[eval].ts:1:22 - error TS7005: Variable 'console' implicitly has an 'any' type.

1 const hello = 'world'console.log(hello)
                       ~~~~~~~
[eval].ts:1:30 - error TS7005: Variable 'log' implicitly has an 'any' type.

1 const hello = 'world'console.log(hello)
                               ~~~
[eval].ts:1:22 - error TS1005: ',' expected.

1 const hello = 'world'console.log(hello)
                       ~~~~~~~
[eval].ts:1:29 - error TS1005: ',' expected.

1 const hello = 'world'console.log(hello)
                              ~
[eval].ts:1:33 - error TS1005: ',' expected.

1 const hello = 'world'console.log(hello)
                                  ~

undefined
```

I would offer to fix this and add test off course, though before I continue I would like to get some feedback on the current approach? If you give me a green light, I will happily continue on this PR :)

blakeembrey commented:
@BrunnerLivio This looks close. The issue appears to be that `appendEval` twice without any new lines is place them directly on top of each other. The new line is probably happening automatically at the end of usual REPL evaluations, so you could try manually appending it for now to see if it fixes things.

BrunnerLivio commented:
@blakeembrey 
Changed the following since the last review:

- Appending `\n` after given "--eval" code (fixes the above described problem)
- Added `-i` alias to `--interactive`

Open questions:
How do I test this addition (if needed)? I don't seem to find any existing test which takes REPL into consideration, which would be required in order to test this addition properly.

blakeembrey commented:
> How do I test this addition (if needed)?

Unfortunately the REPL is under tested, but I'd love to accept any tests any with it! I won't have time to take a look into it unfortunately.

blakeembrey commented:
Merged with https://github.com/TypeStrong/ts-node/commit/20e35b62ad5a4871528a1308d85953b82b8a864a.
